.github/workflows/collector-builder.yml

```
@@ -17,6 +17,7 @@ env:
   COLLECTOR_TAG: ${{ inputs.collector-tag }}
   DEFAULT_BUILDER_TAG: master
   ANSIBLE_CONFIG: ${{ github.workspace }}/ansible/ansible.cfg
+  USE_CCACHE: ${{ !contains(github.event.pull_request.labels.*.name, 'no-ccache') }}
```

Do we want to run with ccache on the master branch and nightlies? That way PRs could just pull the cache from there and see a speed up from the very first commit that gets pushed, otherwise the first build will be as slow as the ones we have now. Maybe we can check if the branch being used is `master` somewhere in our build and clean the cache before building? That way the master branch would always be built from a clean slate, but the cache would still be provided for PRs.
```suggestion
  USE_CCACHE: ${{ github.event_name != 'pull_request' || !contains(github.event.pull_request.labels.*.name, 'no-ccache') }}
```

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#example-of-search-priority

---

Ah good idea. We could skip cache restore on master but always save the cache.